### PR TITLE
AKS - fix editing imported clusters

### LINKED ISSUE #3872: aksv2 - for imported clusters, when performing Edits, UI thinks k8s version has been updated
For imported clusters, when performing Edits from Rancher, UI thinks k8s version has been updated even when thats not the case. 

1. Import AKS cluster
2. Once Active, click Edit
3. Go to NPs. You will see the following even when k8s version has not been changed

Screen Shot 2021-08-13 at 2 25 40 PM

This goes away once k8s version of CP is actually upgraded from Rancher.

(This was a Docker install of v2.6-e22c6bb23ee7e65f04e953afb7a449532cb44db0-head)

Feel free to change the milestone to 2.6.x

### LINKED ISSUE #8501: Imported AKS clusters incorrectly show a K8s upgrade message in the nodepool section, regardless if you have updated the cluster's K8s version
**Setup**
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/2b2ed443065a9f38c4c5764461be33823eb6fec3
- Browser type & version: Chrome

**Describe the bug**

Upon importing a private AKS cluster into Rancher, I wished to upgrade from v1.24.9 to v1.25.5. The upgrade itself was successful, but I noted that when I edited the cluster configuration, I saw this message in the nodepool before I even changed the K8s version:

image

The above message makes sense when I am editing my cluster and I then proceed to update the K8s version. However, I did not update the K8s version at all, but still see this message. As such, if a user has not selected a new K8s version, we shouldn't be showing any message. If a user does select an upgraded K8s version, then we should show this message.

Additionally, I tested provisioning a hosted AKS cluster from Rancher and doing the same workflow, I DO NOT see this issue. After I updated the K8s version, only then did this warning message appear. That is the behavior I would expect, regardless of the AKS cluster being imported or provisioned directly from Rancher.

**To Reproduce**

1. As a standard user, navigate to Cluster Management > Import Existing > Azure AKS.
2. Fill out the requested information and import an existing AKS cluster into Rancher
3. Validate that the cluster and nodes come up as `Active`.
4. Edit the cluster's configuration and scroll down to the node pool section without updating the K8s version for the cluster; you will see this message.

**Result**
Imported AKS clusters that have not updated the K8s versions yet show aforementioned message in the nodepool section.

**Expected Result**

Imported AKS clusters should only show the aforementioned message in the nodepool section when a user has updated the cluster's K8s version.

### LINKED ISSUE #10966: [v2.9] AKS: Unable to edit imported cluster
**Setup**
Rancher version: `v2.9-6d87a11ea46b7571646d7c3d7af704584c39fd62-head`
Browser type & version: Chrome latest

**Describe the bug**
AKS (Vue UI): Unable to manage and edit imported cluster. Most of fields on UI are grayed out and Save button is not enabled. 

**To Reproduce**
- Create AKS cluster and import it from Rancher UI
- Once cluster becomes Active, edit cluster and check the fields

**Screenshots**
image

**Additional Info**
Cluster spec after importing:
```
spec:
  agentImageOverride: ''
  aksConfig:
    authBaseUrl: null
    authorizedIpRanges: null
    azureCredentialSecret: cattle-global-data:cc-897k6
    baseUrl: null
    clusterName: cpinjani-aks1
    dnsServiceIp: null
    dockerBridgeCidr: null
    httpApplicationRouting: null
    imported: true
    kubernetesVersion: null
    loadBalancerSku: null
    logAnalyticsWorkspaceGroup: null
    logAnalyticsWorkspaceName: null
    managedIdentity: null
    monitoring: null
    networkPlugin: null
    networkPolicy: null
    nodePools: null
    outboundType: null
    podCidr: null
    privateCluster: null
    privateDnsZone: null
    resourceGroup: cpinjani-aks
    resourceLocation: eastus
    serviceCidr: null
    subnet: null
    tags: null
    userAssignedIdentity: null
    virtualNetwork: null
    virtualNetworkResourceGroup: null
  answers: {}
  clusterSecrets: {}
  description: ''
  desiredAgentImage: ''
  desiredAuthImage: ''
  displayName: cpinjani-aks1
  dockerRootDir: /var/lib/docker
  enableNetworkPolicy: false
  fleetWorkspaceName: fleet-default
  internal: false
  localClusterAuthEndpoint:
    enabled: false
  windowsPreferedCluster: false
```

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
### Summary
Fixes #10966 
Fixes #3872 
Fixes #8501 

### Occurred changes and/or fixed issues
This PR corrects editing of imported clusters by fixing the upstream syncing logic. Hosted clusters (aks eks gke) track both the configuration managed through Rancher and the configuration managed through their respective cloud provider's platform. You can read more about syncing between the two in the Rancher docs [here](https://ranchermanager.docs.rancher.com/reference-guides/cluster-configuration/rancher-server-configuration/sync-clusters). 

### Areas or cases that should be tested
When testing this PR, it should be noted that once a cluster has been updated through Rancher, it should _only_ be updated through Rancher.
To test issues with the node pool upgrade checkbox/banner stuff:
1. In the azure portal, create an AKS cluster that does not use the latest k8s version available.
2. Import the cluster into Rancher and 'edit config' - the form should be populated and save button not disabled - do _not_ edit the cluster at this stage
3. Go back to the azure portal and upgrade the k8s version of the cluster control plane only (depending on how you made your cluster you may need to disable auto upgrades to do cp-only upgrading)
4. Wait ~5 minutes for the upstream spec to sync
5. Edit the imported cluster through the Rancher UI: there should be an option to upgrade the node pool version
6. Select a new cluster-level k8s version instead: there should now be a banner telling the user they can upgrade the node pool when the cluster upgrade is done
7. Add a new node pool: it should have the same k8s version as the cluster k8s version
8. Save the cluster, let it update, and edit again: there should once again be an option to upgrade the node pool

Other stuff to check:
- Edit an imported cluster and change nothing: the cluster should not go into updating state
- The same edit options should be available for imported and rancher-provisioned AKS clusters

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/utils/kontainer.ts [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+export function syncUpstreamConfig(configPrefix: string, normanCluster: {[key: string]: any}): void {
+  const configKey = `${ configPrefix }Config`;
+  const statusKey = `${ configPrefix }Status`;
+
+  const rancherConfig = normanCluster[configKey] || {};
+  const upstreamConfig = normanCluster?.[statusKey]?.upstreamSpec || {};
+
+  if (!isEmpty(upstreamConfig)) {
+    Object.keys(upstreamConfig).forEach((key) => {
+      if (isEmpty(rancherConfig[key]) && !isEmpty(upstreamConfig.key)) {
```
[the remark this answers is not in the document]
Comment (mantis-toboggan-md) [reply shell/utils/kontainer.ts:20]: Good call - updated now

**Comment by mantis-toboggan-md** [comment]
> I couldn't Import an AKS cluster to test this
> 
> I created a new AKS cluster via porta.azure.com in a new resource group (think i read somewhere only one cluster is allowed by group)
> When using cloud cred from a secret from an app in my azure account, the /meta/aksClusters did not return the cluster
> Question - Did you use the same resource group as a previously created rancher cluster?

I haven't hit this issue before myself. I do typically reuse the same resource group, so I made one in a new group and I do see that cluster available to import, as well as one you created. Was the cluster fully provisioned when you tried to import it?

**Comment by mantis-toboggan-md** [comment]
> Changed Kube version AND added new node. Cluster state did not change for a few minutes before finally going into Updating.
> http put request contained correct aksConfig.kubernetesVersion and aksConfig.nodePools values
> putting this down to a backend issue

I see this as well - it sounds like a backend issue to me. The 5 minute timing to me suggests it may be related to the aks-operator upstream syncing job https://ranchermanager.docs.rancher.com/reference-guides/cluster-configuration/rancher-server-configuration/sync-clusters

> On a broken instance (incorrect resource group) some settings didn't seem to stick
> Updated cluster resource group and location
> http put request contained correct aksConfig.resoourceGroup and aksConfig.resourceLocation
> refreshing the page afterwards showed old values
> Question - have you seen anything like this? It'll probably be a backend issue again

Showing the old values in the form is more concerning to me (the user could possibly save again and overwrite the changes they just made) but I'm having trouble reproducing. What do you mean by an incorrect resource group?

File: shell/utils/kontainer.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (richard-cox) [reply shell/utils/kontainer.ts:20]: i still see this as the old version.

i made the tweak locally and i think it fixed an issue i was seeing (deployed in ukwest, availability zones in edit screen incorrectly shown)

**Review by richard-cox — CHANGES_REQUESTED** [review]
Worked out my issue with seeing the existing cluster to import in our UI, had to explicitly give my app contributor permissions to the subscription.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**